services/provider/server/server_test.go

```diff
@@ -0,0 +1,405 @@
+	assert.NoError(t, client.Create(ctx, monCm))
+	assert.NoError(t, client.Create(ctx, monSc))
+	assert.NoError(t, client.Create(ctx, mgrSvc))
+
+	// When ocsv1alpha1.StorageConsumerStateReady
+	req := pb.StorageConfigRequest{
+		StorageConsumerUUID: string(consumerResource.UID),
+	}
+	storageConRes, err := server.GetStorageConfig(ctx, &req)
+	assert.NoError(t, err)
+	assert.NotNil(t, storageConRes)
+
+	for i := range mockExtR {
+		data, err := json.Marshal(mockExtR[i].Data)
+		assert.NoError(t, err)
+		assert.Equal(t, storageConRes.ExternalResource[i].Data, data)
+		assert.Equal(t, storageConRes.ExternalResource[i].Kind, mockExtR[i].Kind)
+		assert.Equal(t, storageConRes.ExternalResource[i].Name, mockExtR[i].Name)
+	}
+
+	// When ocsv1alpha1.StorageConsumerStateFailed
+	_, err = consumerManager.Create(ctx, "consumer1", "ticket1", resource.MustParse("1G"))
+	assert.NoError(t, err)
+	req = pb.StorageConfigRequest{
+		StorageConsumerUUID: string(consumerResource1.UID),
+	}
+
+	server.consumerManager.nameByUID = map[types.UID]string{
+		consumerResource1.UID: consumerResource1.Name,
+	}
+	storageConRes, err = server.GetStorageConfig(ctx, &req)
+	errCode, _ := status.FromError(err)
+	assert.Error(t, err)
+	assert.Equal(t, errCode.Code(), codes.Internal)
+	assert.Nil(t, storageConRes)
+
+	// When ocsv1alpha1.StorageConsumerConfiguring
+	_, err = consumerManager.Create(ctx, "consumer2", "ticket2", resource.MustParse("1G"))
+	assert.NoError(t, err)
+	req = pb.StorageConfigRequest{
+		StorageConsumerUUID: string(consumerResource2.UID),
+	}
+
+	server.consumerManager.nameByUID = map[types.UID]string{
+		consumerResource2.UID: consumerResource2.Name,
+	}
+	storageConRes, err = server.GetStorageConfig(ctx, &req)
+	assert.Error(t, err)
+	errCode, _ = status.FromError(err)
+	assert.Equal(t, errCode.Code(), codes.Unavailable)
+	assert.Nil(t, storageConRes)
+
+	// When ocsv1alpha1.StorageConsumerDeleting
+	_, err = consumerManager.Create(ctx, "consumer3", "ticket3", resource.MustParse("1G"))
+	assert.NoError(t, err)
+	req = pb.StorageConfigRequest{
+		StorageConsumerUUID: string(consumerResource3.UID),
+	}
+
+	server.consumerManager.nameByUID = map[types.UID]string{
+		consumerResource3.UID: consumerResource3.Name,
+	}
+	storageConRes, err = server.GetStorageConfig(ctx, &req)
+	errCode, _ = status.FromError(err)
+	assert.Error(t, err)
+	assert.Equal(t, errCode.Code(), codes.NotFound)
+	assert.Nil(t, storageConRes)
+
+	// When consumer state is empty
+	_, err = consumerManager.Create(ctx, "consumer4", "ticket4", resource.MustParse("1G"))
+	assert.NoError(t, err)
+	req = pb.StorageConfigRequest{
+		StorageConsumerUUID: string(consumerResource4.UID),
+	}
+	server.consumerManager.nameByUID = map[types.UID]string{
+		consumerResource4.UID: consumerResource4.Name,
+	}
+	storageConRes, err = server.GetStorageConfig(ctx, &req)
+	assert.Error(t, err)
+	errCode, _ = status.FromError(err)
+	assert.Equal(t, errCode.Code(), codes.Unavailable)
+	assert.Nil(t, storageConRes)
```

Comment:
I think the assertion is wrong, if the status has an empty list of CephObjects then there should be no error. 
It all depends on the status of the storage consumer. 

So if StorageConsumer CR is in Ready state and the list is empty it is an OK response (And there should be a test for it)

Reply:
> I think the assertion is wrong, if the status has an empty list of CephObjects then there should be no error. It all depends on the status of the storage consumer.
here I checking for `empty storage consumer status state` which will through `codes.Unavailable`
> 
> So if StorageConsumer CR is in Ready state and the list is empty it is an OK response (And there should be a test for it)
added case when consumer is in ready state but`cephClient` is empty